src/cascadia/WindowsTerminal/WindowThread.cpp

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "WindowThread.h"
+
+WindowThread::WindowThread(const winrt::TerminalApp::AppLogic& logic,
+                           winrt::Microsoft::Terminal::Remoting::WindowRequestedArgs args,
+                           winrt::Microsoft::Terminal::Remoting::WindowManager manager,
+                           winrt::Microsoft::Terminal::Remoting::Peasant peasant) :
+    _peasant{ peasant },
+    _appLogic{ logic },
+    _args{ args },
+    _manager{ manager }
+{
+    // DO NOT start the AppHost here in the ctor, as that will start XAML on the wrong thread!
+}
+
+void WindowThread::Start()
+{
+    _thread = std::thread([this]() {

```

Comment:
You need to `_thread.join()` or ensure that the thread exits before the `WindowThread` is destroyed, otherwise the destructor of `std::thread` will `std::terminate`.

Reply:
Interesting. Not sure the best way to do that. You think I could insert something like a 

```c++
co_await resume_bg();
window->Join(); // calls _thread.join()

co_await wil::resume_foreground... // the rest of the body
```

into the `Exited` handler in `WindowEmperor`?